src/kudu/consensus/leader_election.cc

```diff
@@ -357,30 +366,42 @@ void FlexibleVoteCounter::FetchRegionalUnprunedCounts(
     std::map<std::string, int32_t>* region_unpruned_counts) const {
   CHECK(region_unpruned_counts);
   region_unpruned_counts->clear();
+  std::map<std::string, std::string> uuid_map = GetUuidToQuorumIdOrRegionMap();
   for (const std::pair<const std::string, int64_t>& uuid_pruned_term_pair :
       uuid_to_last_term_pruned_) {
     const std::string& uuid = uuid_pruned_term_pair.first;
     int64_t lpt = uuid_pruned_term_pair.second;
     if (lpt <= term) {
-      const std::string& region = uuid_to_region_.at(uuid);
+      const std::string& region = uuid_map.at(uuid);
       int32_t& region_count = LookupOrInsert(
           region_unpruned_counts, region, 0);
       region_count++;
     }
   }
 }
 
-std::string FlexibleVoteCounter::DetermineRegionForUUID(
+std::string FlexibleVoteCounter::DetermineRegionOrQuorumIdForUUID(
     const std::string& uuid) const {
+  std::map<std::string, std::string> uuid_map = GetUuidToQuorumIdOrRegionMap();
```

Comment:
this is copying the reference from uuid_to_quorum_id_ into a local map for each call.
we can eliminate this expensive copy each time.